compiler/packages/eslint-plugin-react-compiler/src/rules/ReactCompilerRule.ts

```diff
@@ -100,7 +100,6 @@ function makeSuggestions(
 
 const COMPILER_OPTIONS: Partial<PluginOptions> = {
   noEmit: true,
-  compilationMode: "infer",
```

Comment:
Note that `infer` is already the `compilationMode` default, so this override only affects user-specified `compilationMode`s (by overriding all other options to `infer`).